Tests/iaas/security-groups/default-security-group-rules.py

```diff
@@ -47,16 +48,18 @@ def test_rules(cloud_name: str):
         print("No default security group rules defined.")
     else:
         for rule in rules:
-            direction = rule.direction
-            ethertype = rule.ethertype
+            direction = rule["direction"]
+            ethertype = rule["ethertype"]
+            r_custom_sg = rule["used_in_non_default_sg"]
+            r_default_sg = rule["used_in_default_sg"]
             r_custom_sg = rule.used_in_non_default_sg
             r_default_sg = rule.used_in_default_sg
```

Comment:
This looks redundant now.

Reply:
i think i will port the whole part to a function